src/compiler/moduleNameResolver.ts

```diff
@@ -1377,7 +1377,7 @@ namespace ts {
                     trace(state.host, Diagnostics.Trying_substitution_0_candidate_module_location_Colon_1, subst, path);
                 }
                 // A path mapping may have an extension, in contrast to an import, which should omit it.
-                const extension = tryGetExtensionFromPath(candidate);
+                const extension = tryGetExtensionFromPath(subst);
                 if (extension !== undefined) {
                     const path = tryFile(candidate, onlyRecordFailures, state);

```

Comment:
The new code path makes sense, but is this `tryFile` useful anymore? Even if the path pattern specified `.js`, I think I would have expected a colocated `.d.ts` file to be picked up.

Reply:
There are tests to specifically map to a file eg. : https://github.com/microsoft/TypeScript/blob/master/tests/cases/compiler/pathMappingBasedModuleResolution_withExtension.ts#L20